api

```diff
@@ -351,6 +351,38 @@ app_to_pkgname() { #given an app-name, convert it to a unique, valid package-nam
   echo "pi-apps-$(echo "$app" | md5sum | cut -c1-8 | awk '{print $1}')"
 }
 
+add_external_repo() { # add an external apt repo and its gpg key
+  repopath="$1"
+  reponame="$2"
+  pubkeyurl="$3"
```

Comment:
I recommend adding something to check that each of these values are populated, and that the URLs are reachable.

Reply:
I have added a check for each of the variables, as well as a check for if `pubkeyurl` is reachable.

However I'm not sure how we could check if `repopath` is reachable... maybe test if the `Packages` file exists on the server?